Compiling in VSCode fails

I've installed the Powershell Pro Tools extension in latest VSCode as well as the 4.6.2 .NET developer kit and runtime framework. I have the following package.psd1:
```
@{
    Root = 'c:\apps\bin\start-socks.ps1'
    OutputPath = 'c:\apps\bin\out'
    Package = @{
        Enabled = $true
        Obfuscate = $false
        HideConsoleWindow = $true
        PowerShellVersion = '7.4.1'
        DotNetVersion = 'v4.6.2'
        FileVersion = '1.0.0'
        FileDescription = ''
        ProductName = ''
        ProductVersion = ''
        Copyright = ''
        RequireElevation = $false
        ApplicationIconPath = ''
        PackageType = 'Console'
    }
    Bundle = @{
        Enabled = $true
        Modules = $true
        # IgnoredModules = @()
    }
}
        
```
but always get this error when compiling:
```
C:\apps\bin\out\bin\36ca0dc2e46544d88b1341681ffbfa26\ConsolePowerShellHost.cs(3,14): error CS0234: The type or namespace name 'Management' does not exist in the namespace 'System' (are you missing an assembly reference?) [C:\apps\bin\out\bin\36ca0dc2e46544d88b1341681ffbfa26\start-socks.csproj]
C:\apps\bin\out\bin\36ca0dc2e46544d88b1341681ffbfa26\ConsolePowerShellHost.cs(5,17): error CS0234: The type or namespace name 'PowerShell' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?) [C:\apps\bin\out\bin\36ca0dc2e46544d88b1341681ffbfa26\start-socks.csproj]
```
Can anyone please point me in the right direction?

--- a/PowerShellToolsPro.Packager.Test/PowerShellToolsPro.Packager.Test.csproj
+++ b/PowerShellToolsPro.Packager.Test/PowerShellToolsPro.Packager.Test.csproj
@@ -58,6 +58,7 @@
     <Compile Include="ModuleCompilerTests.cs" />
     <Compile Include="OldCompiler.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="ValidateStageTests.cs" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\PowerShellToolsPro.Packager.Test.TestModule\PowerShellToolsPro.Packager.Test.TestModule.csproj">

--- a/PowerShellToolsPro.Packager.Test/ValidateStageTests.cs
+++ b/PowerShellToolsPro.Packager.Test/ValidateStageTests.cs
@@ -1,0 +1,30 @@
+using System;
+using PowerShellToolsPro.Packager;
+using PowerShellToolsPro.Packager.Config;
+using Xunit;
+
+namespace PowerShellToolsPro.Test.Packager
+{
+    public class ValidateStageTests
+    {
+        [Fact]
+        public void ShouldRejectDotNetFrameworkWithPowerShellCore()
+        {
+            var process = new PackageProcess
+            {
+                Config = new PsPackConfig
+                {
+                    Package = new PackageConfig
+                    {
+                        DotNetVersion = "v4.6.2",
+                        PowerShellVersion = "7.4.1"
+                    }
+                }
+            };
+
+            var ex = Assert.Throws<Exception>(() => new ValidateStage().Execute(process, null));
+
+            Assert.Contains(".NET Framework target 'v4.6.2' is not supported with PowerShell '7.4.1'", ex.Message);
+        }
+    }
+}

--- a/PowerShellToolsPro.Packager/PackageProcess.cs
+++ b/PowerShellToolsPro.Packager/PackageProcess.cs
@@ -146,7 +146,7 @@ namespace PowerShellToolsPro.Packager
         {
             if (string.IsNullOrEmpty(process.Config.Package.DotNetVersion)) return;
 
-            var dotNetVersion = process.Config.Package.DotNetVersion.Trim('v').Replace(".", string.Empty);
+            var dotNetVersion = NormalizeDotNetVersion(process.Config.Package.DotNetVersion);
 
             if (!dotNetVersion.StartsWith("net")) dotNetVersion = "net" + dotNetVersion;
 
@@ -181,6 +181,11 @@ namespace PowerShellToolsPro.Packager
             found = false;
             if (process.Config.Package.PowerShellVersion != null)
             {
+                if (IsDotNetFrameworkVersion(dotNetVersion) && process.Config.Package.PowerShellCore)
+                {
+                    throw new Exception($".NET Framework target '{process.Config.Package.DotNetVersion}' is not supported with PowerShell '{process.Config.Package.PowerShellVersion}'. Use Windows PowerShell with .NET Framework or select a .NET Core/.NET 5+ target for PowerShell 7+.");
+                }
+
                 foreach (var version in validNetVersionsForNew)
                 {
                     if (version == dotNetVersion) return;
@@ -191,6 +196,16 @@ namespace PowerShellToolsPro.Packager
                     WriteWarningMessage($"Unvalidated .NET Version specified for selected PowerShell version. Valid values are: {validNetVersionsForNew.Aggregate((x, y) => x + ", " + y)}");
                 }
             }
+        }
+
+        private static string NormalizeDotNetVersion(string dotNetVersion)
+        {
+            return dotNetVersion.Trim('v').Replace(".", string.Empty);
+        }
+
+        private static bool IsDotNetFrameworkVersion(string dotNetVersion)
+        {
+            return dotNetVersion.StartsWith("net4", StringComparison.OrdinalIgnoreCase);
         }
 
         private void ValidatePowerShellVersion(PackageProcess process)
@@ -487,4 +502,3 @@ namespace PowerShellToolsPro.Packager
 
     }
 }
-

